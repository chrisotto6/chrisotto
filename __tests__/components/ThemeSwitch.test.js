import { render, fireEvent } from '@testing-library/react'
import ThemeSwitch from '@/components/ThemeSwitch'

it('renders ThemeSwitch unchanged', () => {
  const { container } = render(<ThemeSwitch />)
  expect(container).toMatchSnapshot()
})

it('handles click of button correctly', () => {
  const { getByTestId, container } = render(<ThemeSwitch />)
  fireEvent.click(getByTestId('theme-button'))
})
