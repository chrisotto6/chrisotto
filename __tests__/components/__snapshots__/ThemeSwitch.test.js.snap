// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ThemeSwitch unchanged 1`] = `
<div>
  <button
    aria-label="Toggle Dark Mode"
    class="ml-1 mr-1 h-8 w-8 rounded p-1 sm:ml-4"
    data-testid="theme-button"
    type="button"
  >
    <svg
      class="text-gray-900 dark:text-gray-100"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.293 13.293A8 8 0 016.707 2.707a8.001 8.001 0 1010.586 10.586z"
        data-testid="light-theme"
      />
    </svg>
  </button>
</div>
`;
